include/grpc/grpc_security.h

```diff
@@ -165,6 +240,23 @@ typedef void (*grpc_credentials_plugin_metadata_cb)(
     void *user_data, const grpc_metadata *creds_md, size_t num_creds_md,
     grpc_status_code status, const char *error_details);
 
+/* Context that can be used by metadata credentials plugin in order to create
+   auth related metadata. */
+typedef struct {
```

Comment:
Why do we need all of this information in the metadata plugin? Why do we need the auth context?